### PR TITLE
Remove a few more usages of thread shared ai and cb

### PR DESCRIPTION
**Pull request by nullkiller**
That should be all. Hope no more usages of thread shared ai and cb left in pathfinding.

### CODE REVIEW

File: AI/VCAI/Pathfinding/AINodeStorage.cpp [hunk 1]
```diff
@@ -185,10 +185,11 @@ std::vector<CGPathNode *> AINodeStorage::calculateNeighbours(
 	return neighbours;
 }
 
-void AINodeStorage::setHero(HeroPtr heroPtr, const CPlayerSpecificInfoCallback * _cb)
+void AINodeStorage::setHero(HeroPtr heroPtr, const CPlayerSpecificInfoCallback * _cb, VCAI * _ai)
```
Comment (alexvins) [inline AI/VCAI/Pathfinding/AINodeStorage.cpp]: Is `_cb` parameter sill needed here? There is one in VCAI.

File: AI/VCAI/Pathfinding/AINodeStorage.h [hunk 2]
```diff
@@ -62,6 +62,7 @@ class AINodeStorage : public INodeStorage
 	/// 1-3 - position on map, 4 - layer (air, water, land), 5 - chain (normal, battle, spellcast and combinations)
 	boost::multi_array<AIPathNode, 5> nodes;
 	const CPlayerSpecificInfoCallback * cb;
+	VCAI * ai;
```
Comment (alexvins) [inline AI/VCAI/Pathfinding/AINodeStorage.h]: Why is it not const? Non const `CGameInterface` instances should be used only for sending request to server

File: AI/VCAI/Pathfinding/AINodeStorage.cpp (hunk 1 above)
Comment (nullkiller) [reply AI/VCAI/Pathfinding/AINodeStorage.cpp]: Ok, will remove

File: AI/VCAI/Goals/VisitHero.cpp [hunk 3]
```diff
@@ -45,7 +45,7 @@ TSubgoal VisitHero::whatToDoToAchieve()
 
 	if(hero && ai->isAccessibleForHero(pos, hero, true) && isSafeToVisit(hero, pos)) //enemy heroes can get reinforcements
 	{
-		if(hero->pos == pos)
+		if(hero->visitablePos() == pos)
```
Comment (nullkiller) [inline AI/VCAI/Goals/VisitHero.cpp]: Small bug fix of 3036

File: AI/VCAI/Pathfinding/AINodeStorage.h (hunk 2 above)
Comment (nullkiller) [reply AI/VCAI/Pathfinding/AINodeStorage.h]: Ok will check

Comment (nullkiller) [reply AI/VCAI/Pathfinding/AINodeStorage.h]: Fixed

**Pull request merged**